docs/Displaying a User Interface/Popup Keyboards/index.md

```diff
@@ -177,10 +185,190 @@ KeyboardListener keyboardListener = new KeyboardListener() {
         // This is sent upon every event, such as keypresses, cancellations, and aborting
     }
+
+    @Override
+    public void onKeyboardDidUpdateInputMask(KeyboardEvent event) {
+        switch (event) {
+            case INPUT_KEY_MASK_ENABLED:
+                // The user enabled input key masking
+                break;
+            case INPUT_KEY_MASK_DISABLED:
+                // The user disabled input key masking
+                break;
+            default:
+                break;
+        }
+    }
 };
 ```
 !@
 
+@![iOS, android, javaSE, javaEE]
+### Configuring the Keyboard Properties
+You can change default keyboard properties by updating !@@![iOS]`sdlManager.screenManager.keyboardConfiguration`!@@![android, javaSE, javaEE]`sdlManager.getScreenManager().setKeyboardConfiguration()`!@@![iOS, android, javaSE, javaEE].!@ 
+
+@![iOS, android, javaSE, javaEE]
+#### Keyboard Language
+You can modify the keyboard language by changing the keyboard configuration's `language`. For example, you can set an `EN_US` keyboard. It will default to `EN_US` if not otherwise set.
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setLanguage(Language.EN_US);
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+@![iOS, android, javaSE, javaEE]
+#### Limited Character List
+You can modify the keyboard to enable only some characters by changing the keyboard configuration's `limitedCharacterList`. For example, you can enable only "a", "b" , and "c" on the keyboard. All other characters will be greyed out (disabled).
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setLimitedCharacterList(Arrays.asList("a", "b", "c"));
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+@![iOS, android, javaSE, javaEE]
+#### Auto Complete List
+You can modify the keyboard to allow an app to pre-populate the text field with a list of suggested or completed entries as the user types by changing the keyboard configuration's `autoCompleteList`. For example, you can pre-populate "test1", "test2", and "test3". 
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setAutoCompleteList(Arrays.asList("test1", "test2", "test3"));
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+@![iOS, android, javaSE, javaEE]
+#### Keyboard Layout (RPC 7.1+)
+You can modify the keyboard layout by changing the keyboard configuration's `keyboardLayout`. For example, you can set a `NUMERIC` keyboard. It will default to `QWERTY` if not otherwise set.
```

Comment:
```suggestion
You can modify the keyboard layout by changing the keyboard configuration's `keyboardLayout`. For example, you can set a `NUMERIC` keyboard. It will default to `QWERTY` if not otherwise set.

!!! NOTE
The numeric keyboard layout is only available on RPC 7.1+. See the section [Checking Keyboard Capabilities](#checking-keyboard-capabilities) to determine if this layout is available.

```